watcher/lib/kapacitor.go

```diff
@@ -0,0 +1,152 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lib
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	client "github.com/influxdata/kapacitor/client/v1"
+	"github.com/influxdata/kapacitor/pipeline"
+	"github.com/influxdata/kapacitor/tick"
+	"github.com/influxdata/kapacitor/tick/stateful"
+)
+
+// KapaClient represents a connection to a kapacitor instance
+type KapaClient interface {
+	CreateTask(opt client.CreateTaskOptions) (client.Task, error)
+	Ping() (time.Duration, string, error)
+}
+
+// NewKapaClient creates a Kapacitor client connection
+func NewKapaClient(url, username, password string) (KapaClient, error) {
+	var creds *client.Credentials
+	if username != "" && password != "" {
+		creds = &client.Credentials{
+			Method:   client.UserAuthentication,
+			Username: username,
+			Password: password,
+		}
+	}
+	clt, err := client.New(client.Config{
+		URL:         url,
+		Credentials: creds,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &PaginatingKapaClient{clt, KapacitorFetchRate}, nil
+}
+
+// ensure PaginatingKapaClient is a KapaClient
+var _ KapaClient = &PaginatingKapaClient{}
+
+// PaginatingKapaClient is a Kapacitor client that automatically navigates
+// through Kapacitor's pagination to fetch all results
+type PaginatingKapaClient struct {
+	KapaClient
+	FetchRate int // specifies the number of elements to fetch from Kapacitor at a time
+}
+
+// KapacitorClient communicates to Kapacitor
+type KapacitorClient struct {
+	kapaClient KapaClient
+}
+
+// NewKapacitorClient creates a client that interfaces with Kapacitor tasks
+func NewKapacitorClient() (*KapacitorClient, error) {
+	username := os.Getenv(KapacitorUsernameEnv)
+	password := os.Getenv(KapacitorPasswordEnv)
+
+	kapaClient, err := NewKapaClient(KapacitorAPIAddress, username, password)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &KapacitorClient{
+		kapaClient: kapaClient,
+	}, nil
+}
+
+func (k *KapacitorClient) Health() error {
+	_, _, err := k.kapaClient.Ping()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (k *KapacitorClient) CreateAlert(name string, script string) error {
+	taskType := client.StreamTask
+	taskTypePipeline := pipeline.StreamEdge
+	if strings.Contains(name, "batch") {
+		taskType = client.BatchTask
+		taskTypePipeline = pipeline.BatchEdge
+	}
+
+	if err := validateTick(script, taskTypePipeline); err != nil {
+		return trace.Wrap(err)
+	}
+
+	dbrps := make([]client.DBRP, 1)
```

Comment:
what is a dbrps?
you can combine declaration with initialization

Reply:
combination of database + retention policy, `CreateTaskOptions` can contain array of `dbrp`, because you can create alert which will stream data from different DBs or from different RPs.